src/datahike/db/utils.cljc

```diff
@@ -146,11 +146,21 @@
   (when eid
     (entid-strict db eid)))
 
+(defn attr-has-ref? [db attr]
+  (and (not (ds/is-system-keyword? attr))
+       (:attribute-refs? (dbi/-config db))
+       (not (nil? attr))))
+
+(defn attr-ref-or-ident [db attr]
+  (if (and (attr-has-ref? db attr)
```

Comment:
Can you swap the two conditionals in the `and` here such that the number case still short circuits? This would save a bit of work in this case.

Reply:
Thanks, I ended up changing order in two places. Good idea!